integration/feature/init/src/MillInitMavenTests.scala

```diff
@@ -0,0 +1,221 @@
+            |[handler] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[example] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[testsuite] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[testsuite-shading] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[transport-blockhound-tests] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |[microbench] dropping classifier ${os.detected.classifier} for dependency io.netty:netty-tcnative:2.0.66.Final
+            |""".stripMargin
+        ) ==> true
+        initRes.out.contains(
+          "generated 47 Mill build file(s)"
+        ) ==> true
+        initRes.isSuccess ==> true
+
+        val resolveRes = eval(("resolve", "_"))
+        Seq(
+          "all",
+          "dev-tools",
+          "common",
+          "buffer",
+          "codec",
+          "codec-dns",
+          "codec-haproxy",
+          "codec-http",
+          "codec-http2",
+          "codec-memcache",
+          "codec-mqtt",
+          "codec-redis",
+          "codec-smtp",
+          "codec-socks",
+          "codec-stomp",
+          "codec-xml",
+          "resolver",
+          "resolver-dns",
+          "resolver-dns-classes-macos",
+          "resolver-dns-native-macos",
+          "transport",
+          "transport-native-unix-common-tests",
+          "transport-native-unix-common",
+          "transport-classes-epoll",
+          "transport-native-epoll",
+          "transport-classes-kqueue",
+          "transport-native-kqueue",
+          "transport-rxtx",
+          "transport-sctp",
+          "transport-udt",
+          "handler",
+          "handler-proxy",
+          "handler-ssl-ocsp",
+          "example",
+          "testsuite",
+          "testsuite-autobahn",
+          "testsuite-http2",
+          "testsuite-osgi",
+          "testsuite-shading",
+          "testsuite-native",
+          "testsuite-native-image",
+          "testsuite-native-image-client",
+          "testsuite-native-image-client-runtime-init",
+          "transport-blockhound-tests",
+          "microbench",
+          "bom"
+        ).forall(resolveRes.out.contains) ==> true
+
+        // additional sources defined by build-helper-maven-plugin
+        val sourcesRes = eval(("show", "transport-native-epoll.sources"))
+        Seq(
+          s"$workspacePath/transport-native-epoll/src/main/java",
+          s"$workspacePath/transport-native-epoll/src/main/c"
+        ).forall(sourcesRes.out.contains) ==> true
+
+        // compile succeeds for submodule
+        val compileRes = eval("codec.compile")
```

Comment:
Let's explicitly list out all the submodules that we expect `.compile` to work on, and loop over them and perform the `compile`, and for those submodules for which `.compile` does not work we can leave a comment about why it currently doesn't work (e.g. `codec.test` due to missing native dependency)